storage/backup/backup-service/src/handlers/utils.rs

```diff
@@ -44,30 +44,101 @@ pub(super) fn reply_with_bcs_bytes<R: Serialize>(
     Ok(Box::new(bytes))
 }
 
+#[must_use]
 pub(super) struct BytesSender {
-    endpoint: &'static str,
-    inner: hyper::body::Sender,
+    buffer: BytesMut,
+    batch_tx: Option<tokio::sync::mpsc::Sender<Bytes>>,
+    sender_task: Option<tokio::task::JoinHandle<(hyper::body::Sender, Result<()>)>>,
 }
 
 impl BytesSender {
-    fn new(endpoint: &'static str, inner: hyper::body::Sender) -> Self {
-        Self { endpoint, inner }
+    fn new(endpoint: &'static str, mut inner: hyper::body::Sender) -> Self {
+        let (batch_tx, mut batch_rx) = tokio::sync::mpsc::channel::<Bytes>(2);
+
+        let sender_task = tokio::spawn(async move {
+            let res = async {
+                while let Some(batch) = batch_rx.recv().await {
+                    let n_bytes = batch.len();
+
+                    inner.send_data(batch).await.into_db_res()?;
+
+                    THROUGHPUT_COUNTER.inc_with_by(&[endpoint], n_bytes as u64);
+                }
+                Ok(())
+            }
+            .await;
+
+            (inner, res)
+        });
+
+        Self {
+            buffer: BytesMut::new(),
+            batch_tx: Some(batch_tx),
+            sender_task: Some(sender_task),
+        }
     }
 
-    async fn send_data(&mut self, chunk: Bytes) -> Result<()> {
-        let n_bytes = chunk.len();
-        self.inner
-            .send_data(chunk)
+    async fn flush_buffer(&mut self) -> Result<()> {
+        self.batch_tx
+            .as_mut()
+            .expect("Batch sender gone.")
+            .send(self.buffer.split().freeze())
             .await
-            .map_err(|e| AptosDbError::Other(e.to_string()))?;
-        THROUGHPUT_COUNTER
-            .with_label_values(&[self.endpoint])
-            .inc_by(n_bytes as u64);
+            .into_db_res()
+    }
+
+    async fn send_data(&mut self, bytes: &[u8]) -> Result<()> {
+        let sender_task = self.sender_task.as_ref().expect("Sender task gone.");
+
+        if sender_task.is_finished() {
+            return Err(AptosDbError::Other(
+                "Sender task finished unexpectedly.".to_string(),
+            ));
+        }
+
+        self.buffer.put_slice(bytes);
+
+        const TARGET_BATCH_SIZE: usize = if cfg!(test) { 10 } else { 1024 * 1024 * 10 };
```

Comment:
Sorry I deleted your comment accidentally @areshand  :trollface:  

But I like constants to be defined where it's used 🧌